Update dendron.ref.cli.md

rlh1994 opened the pull request:
Updated to include all new flags for the commands based on output on `--help` call, and replaced legacy publishing command with new nextjs one. 

Link to publish page not actually tested as I made this change in the github GUI.

Note that I haven't tested the example for the `convert` action on the vault commands section, and did not know what to put for the info and removeCache options for `workspace` so have left these headers commented out for now until they can be populated.

## What does this PR do?
Updates the dendron.ref.cli note to reflect he current flags and publishing options.

## What issues does this PR fix or reference?

### Merge requirements satisfied

> For more information, visit the [Contributing Guide for Documentation](https://wiki.dendron.so/notes/b58801fc-43a9-4d42-a58b-eabc3e8538cb/)

- [ ] Check rendered output to ensure formatting is correct for renderings of wikilinks, note refs, tables, and images. `Dendron: Show Preview` can be used in your workspace to confirm the page renders as expected. Sometimes, `Dendron: Reload Index` needs to be ran if certain wikilinks aren't working as expected. (ignore this if contribution is made via the `Edit this page on GitHub` button from the published site)
- [ ] If this PR includes any refactoring (renaming notes, renaming headers, etc.), make sure that these changes were done via [Dendron Refactoring Commands](https://wiki.dendron.so/notes/srajljj10V2dl19nCSFiC/). This will ensure that any other impacted areas of the documentation, that link to or embed modified notes (via note references), are automatically updated.

Verify GitHub Actions tests are passing, which can be seen in the `Checks` tab of the PR:

- [ ] `URL validator` GitHub Action passes
- [ ] `Dendron site build` GitHub Action passes

> **NOTE:** If you are a new contributor, a maintainer will need to provide approval for GitHub Actions to run on your PRs.

kevinslin reviewed (changes requested):
thanks for the PR!

this is timely since we just updated our guidance around organizing docs, see [[Standard Documentation|dendron://dendron.handbook/leaflet.journal.2022.05.10.standard-documentation]]

specifically, section around CLI here: [[References|dendron://dendron.handbook/leaflet.journal.2022.05.10.standard-documentation.references]]

the gist:
- `cli` for specific commands will go under the matching `dendron.topic` hierarchy 
- for global `cli` options , continue authoring them in `dendron.ref.cli`

heads up that after some discussion, the `find` command from #530 has been renamed to `lookup` (the old lookup renamed to `lookup_legacy`). note that this won't go out until after the june 21st release so you can just leave that out for now

let me know if this makes sense

rlh1994 commented:
That mostly makes sense, although in that case would it be better to place any global options within the `topic.cli` note still so it's all available within a single hierarchy (or I have misunderstood what you meant by options)?

Just more generally on the `ref` vs `topic` guidance, to me it feels a little counter-intuitive  and goes against what I think of with the terms `topic` and `reference`. That leaflet says (from my understanding) to master both the technical details/facts as well as the discussions around them within the `topic` hierarchy, and then to pull out the relevant parts into reference notes in `ref`. I agree with the importance of splitting notes into children for each sub-topic (which will help clean up the CLI note for sure), but would it not be a cleaner approach to master the "facts" and technical details in the `ref` so these can be used when someone needs to just reference the commands, and then embed these via note references into the relevant `topic` or topics they relate to and provide the additional context/discussion around them?  

I'm aware I'm coming at this having really looked at the CLI content which is pretty similar across `ref` and `topic` so it may be I'm missing an obvious reason to go with the topic focused approach.

Either way you importantly only master data in one place which is the main thing, just wanted to share my thoughts and will work on the changes probably early this week 😃

kevinslin commented:
You're right, global options should go into `topic.cli`. i actually forgot we had it as a topic so nice catch 😅

In terms of the ref/topic guidance - there were two ways for us to split it up, either *vertically* (topic.* is the source of truth, hang everything from there and then reference ini refs) or *horizontally* (ref.* is the source of truth and reference everything in topics)
The reason we choose to go with the former is because of ambiguity. when throwing everything into ref, you end up with overlapping commands that still need to be split up. lets take note lookup and schema lookup for example. lets say these are both `cli` commands. if we were to organize them horizontally, we would need some sort of disambiguation between the two lookups (eg. `ref.cli.schema.lookup` or `ref.cli.schema-lookup`). when we group by `topic`, this sort of ambiguity is already ruled out (eg. `dendron.topic.[schema|note].cli.lookup`). like all things, this is a tradeoff - it makes entering a new entry more difficult (you'll need to go deeper into the tree and have an idea of where it needs to go vs just being able to throw something into `ref.cli`) but the result is a *better disambiguated* entry that will be easier to retrieve later on.

all this being said, we also have some proponents of what you suggested so this isn't a done deal. both ways will work, we choose the tradeoff of doing the work upfront for organization but going vertical vs horizontal. as you pointed out, the more important thing is to be consistent. this is where we've chosen to be consistent at but if it turns out the horizontal approach works better in the long term, then we might re-open this discussion. it will also serve as a good litmus test for our refactoring capabilities and being able to handle moving between the horizontal and vertical styles of organizing.

dendron will have done its job as a tool that can enforce flexible structure when it enables both styles of organizing and an *easy* way of migrating between the two when needed. smaller projects benefit from the horizontal scheme due to less overhead, bigger projects benefit from the vertical with more safety. the end documentation would ideally look the same in both cases. there is a similar parallel when organizing code (horizontally across functinality vs vertically by concrete services) where the same tradeoffs are made and evolved.

rlh1994 commented:
That makes more sense to me now, the example helped clear that up for me for sure! I should get the changes done tomorrow or Thursday hopefully, lost track of time the start of the week.

Given the focus on mastering information on `topic`, would there be any value in this being part of the navigation bar on the wiki site? Currently it's only accessible via the search?

rlh1994 commented:
I've moved all the definitions into the respective `topic` hierarchies and created ones where they did not exist, `.ref` now contains only references to `topic` notes.  Please let me know if I've got it right :) 

I've tried to apply a very standard template to each of the notes, leaving headers that aren't relevant at the moment commented out, not sure if you usually do this?

To confirm/do:
- [ ] I have split the `pod` commands into sub-notes (`.topic.pod.import.cli`, `.topic.pod.export.cli`, etc.) then combined them in `topic.pod.cli` is that correct?
- [ ] I've stuck the `dev` command just in the `topic.cli` note - not sure where else to put it?
- [ ] I have left a few options details either commented out or blank where I'm not sure of what might be correct/of value to provide extra context for, these might need populating by someone who uses the CLI more than I do
- [ ] There is an issue I just raised (https://github.com/dendronhq/dendron/issues/3093) that calls out that not all options are being displayed by a commands `--help` flag, I have just provided the output in the notes currently so either this can be merged and those added once that issue is fixed, or wait for that fix if it might be quick
- [ ] The new `lookup` command you mentioned is coming at some point, the existing one will need to be renamed and that new one added.

kevinslin commented:
> Given the focus on mastering information on topic, would there be any value in this being part of the navigation bar on the wiki site? Currently it's only accessible via the search?

Are you talking about dendron.ref.cli? if so, it is available in the nav bar: https://wiki.dendron.so/notes/23a1b942-99af-45c8-8116-4f4bb7dccd21/

>  I have split the pod commands into sub-notes (.topic.pod.import.cli, .topic.pod.export.cli, etc.) then combined them in topic.pod.cli is that correct?

perfect!

>  I've stuck the dev command just in the topic.cli note - not sure where else to put it?

that's fine for now. we'll be creating a `dev` topic in the future but that can be out of scope for this pr

>  I have left a few options details either commented out or blank where I'm not sure of what might be correct/of value to provide extra context for, these might need populating by someone who uses the CLI more than I do

awesome!

>  There is an issue I just raised (--help text does not match full list of arguments for CLI commands dendron#3093) that calls out that not all options are being displayed by a commands --help flag, I have just provided the output in the notes currently so either this can be merged and those added once that issue is fixed, or wait for that fix if it might be quick

ack

>  The new lookup command you mentioned is coming at some point, the existing one will need to be renamed and that new one added.

yep, i'll be adding the docs this week. will wait for this to be merged

kevinslin commented:
this looks great - let me know if you want to flip this from draft -> ready for review. 
only change at this point is to merge this into `dev` instead of master. we generally stage changes in `dev` and then sync docs 2x a week on fridays and tuesdays

rlh1994 commented:
> Are you talking about dendron.ref.cli? if so, it is available in the nav bar: https://wiki.dendron.so/notes/23a1b942-99af-45c8-8116-4f4bb7dccd21/

I meant that none of the `topic` hierarchy is currently visible in the nav bar - you have the `ref` and `concepts` (which seems to be a less complete version of topics from a quick glance?), but `topic` doesn't seem to be published in the nav bar. I feel like more often than not I want the `topic` page as it provides context and more end-user content than the `ref` page which is more just the technical details? For example the [publishing topic](https://wiki.dendron.so/notes/4ushYTDoX0TYQ1FDtGQSg/) I find way more useful and links to more relevant things than the [publishing ref](https://wiki.dendron.so/notes/q960Z622o0RLO32w0B8SP/) most of the time. If `topic` is going to be the core knowledge base is there a case to have that visible in the nav bar?

Only other question is if you like the approach I've taken to the CLI, not sure if it's worth making that an actual template for any future CLI notes?

Otherwise happy for a merge to go ahead!

kevinslin commented:
> I meant that none of the topic hierarchy is currently visible in the nav bar 

ah, gotcha. the reason we didn't have topic in the navbar was to avoid it getting too crowded (hence having them grouped under `user guide`, and then further under `basics/edit/retrieve`) but its awkward and also feels wrong since topic is the central source of truth.

we currently don't have a good solution for this so open to suggestions. we currently have `ref -> topics` as an index for all topics: https://wiki.dendron.so/notes/qW89iJ4vP5UQpC7mRZXKJ/

definitely open to suggestions here.

as for the CLI, i like it and think it'll be the good basis for a template. going to merge this first. if you're okay with creating another PR, we can merge the CLI template in separately :)
